jpos/src/main/java/org/jpos/iso/ISOServer.java

```diff
@@ -700,22 +692,21 @@ public String getCountersAsString () {
 
     public int[] getCounters()
     {
-        Iterator iter = channels.entrySet().iterator();
         int[] cnt = new int[3];
-        cnt[2] = 0;
-        for (int i=0; iter.hasNext(); i++) {
-            Map.Entry entry = (Map.Entry) iter.next();
-            WeakReference ref = (WeakReference) entry.getValue();
-            ISOChannel c = (ISOChannel) ref.get ();
-            if (c != null && !LAST.equals (entry.getKey()) && c.isConnected()) {
-                cnt[2]++;
-                if (c instanceof BaseChannel) {
-                    int[] cc = ((BaseChannel)c).getCounters();
-                    cnt[0] += cc[ISOChannel.RX];
-                    cnt[1] += cc[ISOChannel.TX];
+        channels.entrySet().stream()
+            .filter(ent -> !LAST.equals(ent.getKey()))
+            .filter(ent -> ent.getValue().get().isConnected())
```

Comment:
Remember `ent.getValue().get()` could be null, and the call to `isConnected()` can cause an NPE.